scylla/src/transport/partitioner.rs

```diff
@@ -325,6 +330,76 @@ impl PartitionerHasher for CDCPartitionerHasher {
     }
 }
 
+/// Calculates the token for given prepared statement and serialized values.
+///
+/// Returns the token that would be computed for executing the provided
+/// prepared statement with the provided values.
+pub fn calculate_token(
+    prepared: &PreparedStatement,
+    serialized_values: &SerializedValues,
+) -> Result<Option<Token>, QueryError> {
```

Comment:
Wouldn't it be better if this was in `PreparedStatement`?

`prepared_statement.calculate_token(...)`

```rust
impl PreparedStatement {
  pub fn calculate_token(
      &self,
      serialized_values: &SerializedValues,
  ) -> Result<Option<Token>, QueryError>
}
```